lib/jwt/jwk/key_finder.rb

```diff
@@ -37,22 +36,36 @@ def key_for(kid)
       # Returns the key for the given token
       # @param [JWT::EncodedToken] token the token
       def call(token)
-        key_for(token.header['kid'])
+        kid = token.header['kid']
```

Comment:
If the token happens to have both a kid and for example a x5t header it will never attempt using the x5t. 

Not sure if this is a real scenario and could for sure be a later addition.

Reply:
I think this is a real scenario: here is a payload and header of token from Azure:

```json
[
  {
    "aud": "https://management.core.windows.net",
    "iss": "https://sts.windows.net/57c0dc11-c207-49dd-9bec-76a2d395ea76/",
    "iat": 1745152327,
    "nbf": 1745152327,
    "exp": 1745157876,
    "acr": "1",
    // some more data
  },
  {
    "typ": "JWT",
    "alg": "RS256",
    "x5t": "CNv0OI3RwqlHFEVnaoMAshCH2XE",
    "kid": "CNv0OI3RwqlHFEVnaoMAshCH2XE"
  }
]
```

The keys can be found in https://login.windows.net/common/discovery/keys.